framework/src/media/utils/MediaUtils.cpp

```diff
@@ -53,25 +53,398 @@ audio_type_t getAudioTypeFromPath(std::string datapath)
+		meddbg("Not Supported Format bit/sample : %u\n", bitPerSample);
+		return false;	
+	}
+
+	return true;
+}
+
+bool header_parsing(FILE *fp, audio_type_t audioType, unsigned int *channel, unsigned int *sampleRate, audio_format_type_t *pcmFormat)
+{
+	unsigned char *header;
+	unsigned char tag[2];
+	bool isHeader;
+
+	switch (audioType) {
+	case AUDIO_TYPE_MP3:
+		isHeader = false;
+		while (fread(tag, sizeof(unsigned char), 1, fp) > 0) {
+			if (tag[0] == 0xFF) {
+				isHeader = true;
+				break;
+			}
+		}
+		if (isHeader) {
+			header = (unsigned char *)malloc(sizeof(unsigned char) * (MP3_HEADER_LENGTH + 1));
+			if (header == NULL) {
+				medvdbg("malloc failed error\n");
+				return false;
+			}
+
+			if (fseek(fp, -1, SEEK_SET) != 0) {
+				meddbg("file seek failed error\n");
+				return false;
+			}
+			if (fread(header, sizeof(unsigned char), MP3_HEADER_LENGTH, fp) != MP3_HEADER_LENGTH || !mp3_header_parsing(header, channel, sampleRate)) {
+				free(header);
+				return false;
+			}
+
+		} else {
+			medvdbg("no header\n");
+			return false;
+		}
+		break;
+	case AUDIO_TYPE_AAC:
+		isHeader = false;
+		while (fread(tag, sizeof(unsigned char), 1, fp) > 0) {
+			if (tag[0] == 0xFF) {
+				isHeader = true;
+				break;
+			}
+		}
+		if (isHeader) {
+			header = (unsigned char *)malloc(sizeof(unsigned char) * (AAC_HEADER_LENGTH + 1));
+			if (header == NULL) {
+				medvdbg("malloc failed error\n");
+				return false;
+			}
+
+			if (fseek(fp, -1, SEEK_SET) != 0) {
+				meddbg("file seek failed error\n");
+				return false;
+			}
+
+			if (fread(header, sizeof(unsigned char), AAC_HEADER_LENGTH, fp) == AAC_HEADER_LENGTH && !aac_header_parsing(header, channel, sampleRate)) {
+				free(header);
+				return false;
+			}
+		} else {
+			medvdbg("no header\n");
+			return false;
+		}
+		break;
+	case AUDIO_TYPE_WAVE:
+		header = (unsigned char *)malloc(sizeof(unsigned char) * (WAVE_HEADER_LENGTH + 1));
+		if (header == NULL) {
+			medvdbg("malloc failed error\n");
+			return false;
+		}
+
+		if (fread(header, sizeof(unsigned char), WAVE_HEADER_LENGTH, fp) == WAVE_HEADER_LENGTH && !wave_header_parsing(header, channel, sampleRate, pcmFormat)) {
+			free(header);
+			return false;
+		}
+		break;
+	default:
+		medvdbg("does not support header parsing\n");
+		return false;
+	}
+
+	if (header != NULL) {
+		free(header);
+	}
+
+	if (fseek(fp, 0, SEEK_SET) != 0) {
+		meddbg("file seek failed error\n");
+		return false;
+	}
+	
+	return true;
+}
+
+bool header_parsing(unsigned char *buffer, unsigned int bufferSize, audio_type_t audioType, unsigned int *channel, unsigned int *sampleRate, audio_format_type_t *pcmFormat)
+{
+	int headPoint;
+	unsigned char *header;
+	bool isHeader;
+	switch (audioType) {
+	case AUDIO_TYPE_MP3:
+		isHeader = false;
+		headPoint = 0;
+		while (headPoint < bufferSize) {
+			if (buffer[headPoint] == 0xFF) {
+				isHeader = true;
+				break;
+			}
+			headPoint++;
+		}
+		if (isHeader && MP3_HEADER_LENGTH <= bufferSize - headPoint) {
```

Comment:
if buffer size is smaller than header, we can't get data properly.
so first of all, we should check this and if it is smaller than header size, return error needed.
So, consider expandability,  we had better provide getHeaderInfo (to get size...and so on)later, then application can realize minimum size of buffer.
so please do below
1.  Change logic that size checking code 
2. Put TODO about expandability.